test/support/page_objects/discover_page.js

```diff
@@ -285,4 +285,55 @@ export default class DiscoverPage {
       .catch(() => false);
   }
 
+  clickFieldListItem(field) {
+    return this.findTimeout
+    .findByCssSelector('li[attr-field="' + field + '"]').click();
+  }
+
+  clickFieldListItemAdd(field) {
+    return this.findTimeout
+    .findByCssSelector('li[attr-field="' + field + '"] > div > button').click();
+  }
+
+  clickFieldListItemVisualize(field) {
+    return this.findTimeout
+    .findByCssSelector('li[attr-field="' + field + '"] > a').click();
+  }
```

Comment:
This and the method above feel a little too generic, I'd worry about them breaking if we add another link or button. Maybe add data-test-subj attributes for these too?